### PR TITLE
Core 1.47

### PR DESCRIPTION
**Pull request by Simon-Laux**
upgrade core and adjust api to the breaking changes and addition of quotes.

I haven't adjusted the tests, yet, they seem to run fine anyway... seriously we gotta rethink the testing for this project.[with the changes I made the tests should be broken] (converting it to typescript could be a first step and then possibly moving it to a different test framework)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
 
+## [1.46.0] - 2020-21-10
+
+Breaking changes:
+- `deltachat.updateDeviceChats()` removed
+  this is now done automatically during configure
+  unless the new config-option `bot` is set deltachat-core-rust/#1957
+- `deltachat.markNoticedAllChats()` removed
+- `chat.isVerified()` is replaced by `chat.isProtected()`
+- `deltachat.createUnverifiedGroupChat(chatName)` and `deltachat.createVerifiedGroupChat(chatName)` are replaced by a single function `deltachat.createGroupChat(chatName, is_protected)`
```
Comment (hpk42) [inline CHANGELOG.md:11]: Why these two naming changes?  I don't see how they help help with things while it immediately breaks desktop.

**Review by hpk42 — APPROVED** [review]
let's merge this despite my comments. not worth discussing and postponing because of it.  
i suggest to try get a working node/desktop/core master configuration and keep it working together until gplay android is published.  Note that core47 is not stable yet.

File: CHANGELOG.md (hunk 1 above)
Comment (Simon-Laux) [reply CHANGELOG.md:11]: It is broken anyway in multiple ways, this is one of two parts, part two is an desktop pr that uses this pr.

**Comment by Simon-Laux** [comment]
somehow I still get this error when trying to set a quote:
`DC_EVENT_WARNING 0 deltachat-ffi/src/lib.rs:3312: failed to set quote: Message without Message-Id cannot be quoted`

File: lib/message.ts [hunk 2]
```diff
@@ -284,6 +292,10 @@ export class Message {
     return this
   }
 
+  setQuote(qoutedMessage: Message) {
```
Comment (link2xt) [inline lib/message.ts:295]: should be `quoted`

File: src/module.c [hunk 3]
```diff
@@ -2236,6 +2254,17 @@ NAPI_METHOD(dcn_msg_set_file) {
   NAPI_RETURN_UNDEFINED();
 }
 
+NAPI_METHOD(dcn_msg_set_quote) {
+  NAPI_ARGV(2);
+  dc_msg_t* dc_msg;
+  NAPI_STATUS_THROWS(napi_get_value_external(env, argv[0], (void**)&dc_msg));
+  dc_msg_t* dc_msg_qoute;
```
Comment (link2xt) [inline src/module.c:2261]: `quote`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+## [1.47.0] - 2020-26-10
+
+Breaking changes:
+- `deltachat.updateDeviceChats()` removed
+  this is now done automatically during configure
+  unless the new config-option `bot` is set deltachat-core-rust/#1957
+- `deltachat.markNoticedAllChats()` removed
+- `chat.isVerified()` is replaced by `chat.isProtected()`
+- `deltachat.createUnverifiedGroupChat(chatName)` and `deltachat.createVerifiedGroupChat(chatName)` are replaced by a single function `deltachat.createGroupChat(chatName, is_protected)`
+- if an message has a qoute to another message you need to use `message.getQuotedText()` to get the quoted message
```
Comment (link2xt) [inline CHANGELOG.md:12]: same typo, quote

File: src/module.c [hunk 5]
```diff
@@ -2236,6 +2254,17 @@ NAPI_METHOD(dcn_msg_set_file) {
...
+  NAPI_STATUS_THROWS(napi_get_value_external(env, argv[1], (void**)&dc_msg_qoute));
```
Comment (link2xt) [inline src/module.c]: Could be wrapped into `NAPI_ARGV_DC_MSG` macro and replaced with
```
NAPI_ARGV_DC_MSG(dc_msg, 0);
NAPI_ARGV_DC_MSG(dc_msg, 1);
```

**Comment by Simon-Laux** [comment]
the tests seem to be still running, I'll make an issue for fixing them and merge this so we can continue with an 1.48 core pr.

File: lib/message.ts (hunk 2 above)
Comment (link2xt) [reply lib/message.ts:295]: Seems not to be changed, marked as unresolved.

**Review by link2xt — APPROVED** [review]

**Pull request merged**